perceval/simulators/simulator.py

```diff
@@ -300,31 +300,40 @@ def _probs_svd_generic(self, input_dist, p_threshold, progress_callback: Optiona
     def _probs_svd_fast(self, input_dist, p_threshold, progress_callback: Optional[Callable] = None):
         decomposed_input = []
         """decomposed input:
-       From a SVD = {
-           bs_1: p1,
-           bs_2: p2,
-           ...
-           bs_k: pk
-       }
-       the following data structure is built:
-       [
-           (p1, [bs_1,]),
-           ...
-           (pk, [bs_k,])
-       ]
-       where [bs_x,] is the list of the un-annotated separated basic state (bs_x.separate_state())"""
+           From a SVD = {
+               bs_1: p1,
+               bs_2: p2,
+               ...
+               bs_k: pk
+           }
+           the following data structure is built:
+           [
+               (p1, [bs_1,]),
+               ...
+               (pk, [bs_k,])
+           ]
+           where [bs_x,] is the list of the un-annotated separated basic state (result of bs_x.separate_state())
+        """
         for sv, prob in input_dist.items():
             if max(sv.n) >= self._min_detected_photons:
                 decomposed_input.append(
                     (prob, sv[0].separate_state(keep_annotations=False))
                 )
             else:
                 self._physical_perf -= prob
-        input_set = set([state for s in decomposed_input for state in s[1]])
+
+        """Create a cache with strong simulation of all unique input"""
         cache = {}
-        for state in input_set:
+        input_set = set([state for s in decomposed_input for state in s[1]])
+        len_input_set = len(input_set)
+        for idx, state in enumerate(input_set):
             self._backend.set_input_state(state)
             cache[state] = self._backend.prob_distribution()
+            if progress_callback and idx % 10 == 0:
```

Comment:
Did you have time to check this part or do you want me to do it ?

Reply:
Yes, I'm checking this a lot on the cloud worker side